kms/keysource.go

```diff
@@ -183,12 +183,24 @@ func (key MasterKey) createStsSession(config aws.Config, sess *session.Session)
 
 func (key MasterKey) createSession() (*session.Session, error) {
 	re := regexp.MustCompile(`^arn:aws[\w-]*:kms:(.+):[0-9]+:(key|alias)/.+$`)
+	// hmm, have to change arn? this is ugly but we must have the region
+	// implies us-east-1:alias/my-alias-name
+	re_alias := regexp.MustCompile(`^(.+):alias/.+$`)
+
 	matches := re.FindStringSubmatch(key.Arn)
-	if matches == nil {
-		return nil, fmt.Errorf("No valid ARN found in %q", key.Arn)
-	}
+	alias_matches := re_alias.FindStringSubmatch(key.Arn)
+
+	if matches != nil {
+		config := aws.Config{Region: aws.String(matches[1])}
+	} else {
 
-	config := aws.Config{Region: aws.String(matches[1])}
+		if alias_matches != nil {
+			// its an alias the region is irrelevant
+			config := aws.Config{Region: aws.String(matches[0])}
+		} else {
+			return nil, fmt.Errorf("No valid ARN found in %q", key.Arn)
+		}
+	}
```

Comment:
Unnest this into an else if and an else.

Reply:
Thanks! I'm not sure about my butchering the arn pattern to force the region into the alias?

Reply:
Have you tested what happens if you provide a bogus region to the alias? e.g. `blah:alias/foo`.

I don't currently have an AWS account set up so this is not particularly easy for me to test :(